extensions/controllers/sandboxclaim_controller.go

```diff
@@ -76,26 +81,68 @@ func (r *SandboxClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
-	// cache the original status from sandboxclaim
+	// Cache original status
 	originalClaimStatus := claim.Status.DeepCopy()
+
 	var err error
 	var sandbox *v1alpha1.Sandbox
 	var template *extensionsv1alpha1.SandboxTemplate
 
-	// Try getting template
-	if template, err = r.getTemplate(ctx, claim); err == nil || k8errors.IsNotFound(err) {
-		// This ensures the firewall is up before the pod starts.
-		if npErr := r.reconcileNetworkPolicy(ctx, claim, template); npErr != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to reconcile network policy: %w", npErr)
+	// Check Expiration
+	// The Claim is the source of truth for its own expiration logic.
+	claimExpired := false
+	if claim.Spec.Lifecycle != nil && claim.Spec.Lifecycle.ShutdownTime != nil {
+		claimExpired = time.Now().After(claim.Spec.Lifecycle.ShutdownTime.Time)
+	}
+
+	// 1. Manage Sandbox Resources
+	if !claimExpired {
+		// Logic for active Claims
+		if template, err = r.getTemplate(ctx, claim); err == nil || k8errors.IsNotFound(err) {
+			// Ensure NetworkPolicy
+			if npErr := r.reconcileNetworkPolicy(ctx, claim, template); npErr != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to reconcile network policy: %w", npErr)
+			}
+			sandbox, err = r.getOrCreateSandbox(ctx, claim, template)
+		}
+	} else {
+		// Logic for expired Claims
+		// We only check if the Sandbox exists to update status.
+		// If Policy is Delete, we don't need to check Sandbox status.
+		// We just delete the Claim immediately.
+		if claim.Spec.Lifecycle != nil && claim.Spec.Lifecycle.ShutdownPolicy == extensionsv1alpha1.ShutdownPolicyDelete {
+			log.Info("Deleting Claim because ShutdownPolicy=Delete and time has expired")
+			if r.Recorder != nil {
+				r.Recorder.Event(claim, corev1.EventTypeNormal, extensionsv1alpha1.ClaimExpiredReason, "Deleting Claim (ShutdownPolicy=Delete)")
+			}
+			// We ignore NotFound because if it's already gone.
+			if delErr := r.Delete(ctx, claim); delErr != nil {
+				return ctrl.Result{}, client.IgnoreNotFound(delErr)
+			}
+			return ctrl.Result{}, nil
 		}
 
-		// Try getting sandbox even if template is not found
-		// It is possible that the template was deleted after the sandbox was created
-		sandbox, err = r.getOrCreateSandbox(ctx, claim, template)
+		// If we reached here, Policy is "Retain" (or nil).
+		// Check Sandbox existence only to update the Status.
+		existingSandbox := &v1alpha1.Sandbox{}
+		if err = r.Get(ctx, req.NamespacedName, existingSandbox); err == nil {
```

Comment:
Do you need `existingSandbox`? Can you just use `sandbox`?  Did you handle the case when err != nil?